example/server.js

```diff
@@ -2,22 +2,51 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var tags = require('../lib/index.js');
+var fs = require('fs');
+var pg = require('pg');
 
 var server = new Hapi.Server();
+var tagsData = JSON.parse(fs.readFileSync('./example/tags.json').toString());
+var tags = require('../lib/index.js');
+var tagsPool = new pg.Pool();
+
+
+function init (port, callback) {
+  server.connection({ port: port });
+
+  server.register([{
+    register: tags,
+    options: { tags: tagsData, tagsPool: tagsPool }
+  }], function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    server.route([{
+      method: 'GET',
+      path: '/',
+      handler: function (request, reply) {
+        tagsPool.connect(function (connErr, client, done) {
+          Hoek.assert(!connErr, connErr);
+          client.query('select * from tags', function (dberr, res) {
+            Hoek.assert(!dberr, dberr);
+            done();
 
-server.connection({ port: process.env.PORT || 3000 });
+            return reply(res.rows);
+          });
+        });
+      }
+    }, {
+      method: 'GET',
+      path: '/hello',
+      handler: function (request, reply) {
+        reply('hello');
+      }
+    }]);
 
-server.register(tags, function (err) {
-  Hoek.assert(!err, 'error registering plugin');
-});
+    return callback(null, server);
+  });
+}
 
-server.route({
-  method: 'GET',
-  path: '/',
-  handler: function (request, reply) {
-    reply(request.jackmisawesome);
-  }
-});
 
-module.exports = server;
+module.exports = { init: init, tagsPool: tagsPool };
```

Comment:
It's true that we might need to refactor some tests, you can try to expose the pool on all the application at the moment and gradually replacing the tests. It might break a few tests but I think it will be more easier on a long term to create the pool inside init and to pass it to the plugins. But yes step by step :+1: